package.json

```diff
@@ -172,5 +172,33 @@
     "urijs": "^1.19.2",
     "vkbeautify": "^0.99.3",
     "xml2js": "^0.4.23"
+  },
```

Comment:
Checking my understanding.  These resolutions put our thumb on the scale so that yarn resolves to at least the version specified.

Reply:
Yes - this mechanism basically allows you to specify the SemVer for nested dependencies without editing the yarn.lock file directly.

Resolutions in the form `**/<package>` enforce a range _wherever_ the package is located in the tree, whereas those with a more explicit path target particular nested dependencies.

So for example, in the case of Acorn, we actually need 3 versions. We have 7.x in the main dependencies, but the Embroider package with Acorn as a subdependency requires 6.x, so the selector there is forcing it to use the version with the vulnerability patched, overriding the SemVer range in its own package.json which yarn.lock was satisfying (and hence keeping us using a vulnerable version).

There are some other packages in the tree that currently use Acorn 5.7.3, but they break when bumped to Acorn 5.7.4 (sadly, as that's the version with the vuln fix), but if they _did_ work with 5.7.4 but not 6.x, extra lines could be added to force that to happen.